cmd/registry/cmd/label.go

```diff
@@ -0,0 +1,258 @@
+		for k, v := range labelPairs {
+			if v == "" {
+				delete(task.version.Labels, k)
+			} else {
+				task.version.Labels[k] = v
+			}
+		}
+		req := &rpc.UpdateApiVersionRequest{
+			ApiVersion: task.version,
+			UpdateMask: &field_mask.FieldMask{
+				Paths: []string{"labels"},
+			},
+		}
+		_, err := task.client.UpdateApiVersion(task.ctx, req)
+		return err
+	}
+	return nil
+}
+
+func (task *labelSpecTask) Name() string {
+	return "label " + task.spec.Name
+}
+
+func (task *labelSpecTask) Run() error {
+	if task.spec.Labels == nil {
+		task.spec.Labels = make(map[string]string)
+	}
+	if len(labelPairs) > 0 {
+		log.Printf("label %s with %v", task.spec.Name, labelPairs)
+		for k, v := range labelPairs {
+			if v == "" {
+				delete(task.spec.Labels, k)
+			} else {
+				task.spec.Labels[k] = v
+			}
+		}
+		req := &rpc.UpdateApiSpecRequest{
+			ApiSpec: task.spec,
+			UpdateMask: &field_mask.FieldMask{
+				Paths: []string{"labels"},
+			},
+		}
+		_, err := task.client.UpdateApiSpec(task.ctx, req)
+		return err
+	}
+	return nil
+}
+
+func matchAndHandleLabelCmd(
+	ctx context.Context,
+	client connection.Client,
+	taskQueue chan core.Task,
+	name string,
+) error {
+	// First try to match collection names.
+	if m := names.ApisRegexp().FindStringSubmatch(name); m != nil {
+		return labelAPIs(ctx, client, m, labelFilter, taskQueue)
+	} else if m := names.VersionsRegexp().FindStringSubmatch(name); m != nil {
+		return labelVersions(ctx, client, m, labelFilter, taskQueue)
+	} else if m := names.SpecsRegexp().FindStringSubmatch(name); m != nil {
+		return labelSpecs(ctx, client, m, labelFilter, taskQueue)
+	}
+
+	// Then try to match resource names.
+	if m := names.ApiRegexp().FindStringSubmatch(name); m != nil {
+		return labelAPIs(ctx, client, m, labelFilter, taskQueue)
+	} else if m := names.VersionRegexp().FindStringSubmatch(name); m != nil {
+		return labelVersions(ctx, client, m, labelFilter, taskQueue)
+	} else if m := names.SpecRegexp().FindStringSubmatch(name); m != nil {
+		return labelSpecs(ctx, client, m, labelFilter, taskQueue)
+	} else {
+		return fmt.Errorf("unsupported resource name %s", name)
+	}
+}
+
+func labelAPIs(
+	ctx context.Context,
+	client *gapic.RegistryClient,
+	segments []string,
+	filterFlag string,
+	taskQueue chan core.Task) error {
+	return core.ListAPIs(ctx, client, segments, filterFlag, func(api *rpc.Api) {
+		taskQueue <- &labelApiTask{
+			ctx:    ctx,
+			client: client,
+			api:    api,
+		}
+	})
+}
+
+func labelVersions(
+	ctx context.Context,
+	client *gapic.RegistryClient,
+	segments []string,
+	filterFlag string,
+	taskQueue chan core.Task) error {
+	return core.ListVersions(ctx, client, segments, filterFlag, func(version *rpc.ApiVersion) {
+		taskQueue <- &labelVersionTask{
+			ctx:     ctx,
+			client:  client,
+			version: version,
+		}
+	})
+}
+
+func labelSpecs(
+	ctx context.Context,
+	client *gapic.RegistryClient,
+	segments []string,
+	filterFlag string,
+	taskQueue chan core.Task) error {
```

Comment:
Nit: I don't think this is better than a single line, but that's subjective of course. Is there a particular line length you're formatting to?

Reply:
Not specifically - my editor window is usually 80-100 characters wide, and these lines would be around 140 chars.